pool.go

```diff
@@ -46,8 +38,35 @@ func (p *Pool) SetStrategy(strategy Strategy) error {
 	return nil
 }
 
+// Add .
+func (p *Pool) Add(ns Namespace) {
+	s := &ring.Ring{Value: ns}
+	r := p.Nodes.Link(s)
+	s.Link(r)
+}
+
+// Drop .
+func (p *Pool) Drop(domain string) error {
+	for i := 0; i < p.Nodes.Len(); i++ {
+		ns, ok := p.Nodes.Value.(Namespace)
```

Comment:
most likely not, but since `p.Nodes` is exported, it's possible that the caller did something bad with it.